src/AutoMapperInterface.php

```diff
@@ -24,4 +24,16 @@ interface AutoMapperInterface
      * @return ($target is class-string|Target ? Target|null : array<mixed>|null)
      */
     public function map(array|object $source, string|array|object $target, array $context = []): array|object|null;
+
+    /**
+     * @template Source of object
+     * @template Target of object
+     *
+     * @param array<array<mixed>>|array<Source> $collection
+     * @param class-string<Target>|'array'      $target
+     * @param MapperContextArray                $context
+     *
+     * @return ($target is class-string ? array<Target> : array<mixed>)
+     */
+    public function mapCollection(array $collection, string $target, array $context = []): array;
```

Comment:
maybe we could map any iterable?
```suggestion
    public function mapCollection(iterable $collection, string $target, array $context = []): array;
```